frontend/static/scripts/main.js

```diff
@@ -86,19 +86,85 @@ function gen_counterexamples_html(counterexamples){
 
 $(function () {
 
-    if (!Cookies.get('username')) {
-        $('#myModal').modal('show');
+    if (!Cookies.get('token')) {
+        $('#myModal').modal({backdrop: 'static', keyboard: false});
     }
 
-    $('#myModal').on('hidden.bs.modal', function (e) {
-        if ($('#username-input').val() == "" || $('#institution-input').val() == "") {
-            $('#myModal').modal('show');
+    $('.register-btn').click(function() {
+        var name = $('#name-input').val();
+        var email = $('#email-input').val();
+        var pass = $('#pass-input').val();
+        var inst = $('#institution-input').val();
+
+        if (!name || !email || !pass || !inst) {
+            return false;
+        }
+
+        $.ajax({
+            url: '/register',
+            method: 'POST',
+            data: {
+                "name": name,
+                "email": email,
+                "password": pass,
+                "institution": inst
+            },
+            success: function (data) {
+                var res = $.parseJSON(data)
+                var stat = res['status'];
+                if (stat == 1) {
+                    console.log('email taken');
+                    $('.email-taken').show();
+                } else if (stat == 0) {
+                    Cookies.set('token', res['token']);
+                    $('#myModal').modal('hide');
```

Comment:
`myModal` is not a very good name here. Can you change it to a name that makes more sense?